pkg/ipcache/metadata.go

```diff
@@ -427,22 +427,37 @@ func (ipc *IPCache) resolveIdentity(ctx context.Context, prefix netip.Prefix, in
 	}
 
 	lbls := info.ToLabels()
+	hasRemoteNode := lbls.Has(labels.LabelRemoteNode[labels.IDNameRemoteNode])
+	hasLocalHost := lbls.Has(labels.LabelHost[labels.IDNameHost])
+
+	// Add in CIDR labels where needed:
+	// - for out-of-cluster entities
+	// - for remote nodes, when policy-cidr-selects-nodes is enabled
+	// - for localhost, when policy-cidr-selects-nodes is enabled
+	if !lbls.Has(labels.LabelHealth[labels.IDNameHealth]) &&
+		!lbls.Has(labels.LabelIngress[labels.IDNameIngress]) &&
+		(!hasRemoteNode || (option.Config.PolicyCIDRSelectsNodes && hasRemoteNode)) &&
+		(!hasLocalHost || (option.Config.PolicyCIDRSelectsNodes && hasLocalHost)) {
+		cidrLabels := cidrlabels.GetCIDRLabels(prefix)
+		lbls.MergeLabels(cidrLabels)
```

Comment:
If we allocate the node identities the standard way using the old API via `AllocateCIDRs()` which calls into this function, I suspect we could drop the entire last two commits and just handle the remote node label restore logic right here.

```suggestion
		lbls.MergeLabels(cidrLabels)
                if option.Config.PolicyCIDRSelectsNodes &&
                  restoredIdentity.Scope() == identity.IdentityScopeRemoteNode &&
                  !hasRemoteNode && {
                       lbls.MergeLabels(labels.LabelRemoteNode)
                       hasRemoteNode = true
               }
```

Reply:
yes, but only for the restore case, not for the new-node case.

Reply:
@squeed @joestringer The updated logic does not include logic for restoring host CIDR labels, which the referenced commit(s) contained. Why was it removed?